File: programming/Backends/services/NotificationService/src/main/java/org/tongji/sse/entity/Notification.java
package org.tongji.sse.entity;

import jakarta.persistence.*;
import lombok.*;
import org.tongji.sse.type.NotificationType;
import org.tongji.sse.type.TargetType;

import java.time.Instant;

@Entity
@Table(name = "notification", indexes = {
        @Index(columnList = "eventId", unique = true)
})
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String eventId;      // 对应 NotificationEvent.eventId，用于幂等

    @Column(nullable = false)
    private Long receiverUserId;

    @Column(nullable = false)
    private Long actorUserId;

    @Column(nullable = false)
    private NotificationType type;         // LIKE / COMMENT / REPLY

    @Column(nullable = false)
    private String content;

    @Column(nullable = false)
    private TargetType targetType;   // AUDIO / COMMENT

    @Column(nullable = false)
    private Long targetId;

    @Column(nullable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Boolean isRead;
}

File: programming/Frontends/src/types/Notification.ts
export enum NotificationType {
  LIKE = 'LIKE',
  COMMENT = 'COMMENT',
  REPLY = 'REPLY'
}

export enum TargetType {
  AUDIO = 'AUDIO',
  COMMENT = 'COMMENT',
  REPLY = 'REPLY'
}

export interface NotificationDTO {
  id: number;
  receiverUserId: number;
  actorUserId: number;
  type: NotificationType;
  content: string;
  targetType: TargetType;
  targetId: number;
  createdAt: string; // ISO 8601 字符串，如 "2025-04-01T12:00:00Z"
  isRead: boolean; // 是否已读
}
